pkcs11/pkcs11_test.c

```diff
@@ -486,6 +486,11 @@ int ecdh_test(ecc_key* privKey, ecc_key* pubKey, int check)
     byte   out[256/8];
     word32 outSz = sizeof(out);
 
+#if defined(ECC_TIMING_RESISTANT) && (!defined(HAVE_FIPS) || \
+    (!defined(HAVE_FIPS_VERSION) || (HAVE_FIPS_VERSION != 2))) && \
+    !defined(HAVE_SELFTEST)
+    ret = wc_ecc_set_rng(privKey, &rng);
```

Comment:
Can you explain why we only set the RNG in some cases?

Reply:
The functions was added in the last release and is required for the secret generation function when the timing resistant implementation is compiled in.
Old FIPS code doesn't have the API and has an implementation that doesn't required the random so we can't call it.

Reply:
That makes sense, thanks!